In [ ]:
import pandas as pd

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving data.csv to data (3).csv


In [ ]:
import io
df = pd.read_csv(io.BytesIO(uploaded['data.csv']))

In [ ]:
df.head()

Subject_ID  Age_X  Glucose_Level_Y
0           1     43               99
1           2     21               65
2           3     25               79
3           4     42               75
4           5     57               87

In [ ]:
df.corr()

Subject_ID     Age_X  Glucose_Level_Y
Subject_ID         1.000000  0.695582        -0.130664
Age_X              0.695582  1.000000         0.529809
Glucose_Level_Y   -0.130664  0.529809         1.000000

In [ ]:
import numpy as np

In [ ]:
npArr1 = np.array(df['Age_X'])
npArr2 = np.array(df['Glucose_Level_Y'])

In [ ]:
from scipy.stats import pearsonr

In [ ]:
pearsonr(df['Age_X'], df['Glucose_Level_Y'])

(0.5298089018901743, 0.27964465700487184)

In [ ]:
# !pip install pingouin

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 185 kB 7.3 MB/s 
     |████████████████████████████████| 9.8 MB 8.6 MB/s 
  Created wheel for pingouin: filename=pingouin-0.5.2-py3-none-any.whl size=196206 sha256=94308d717112fe716ae6062ba3edbf3202895839f641894c37d916693d7eb472
  Stored in directory: /root/.cache/pip/wheels/11/5a/63/a6d32fc26fa462c731f65480bfb98ff7bd39b8ebcb4bc6c2fe
  Created wheel for littleutils: filename=littleutils-0.2.2-py3-none-any.whl size=7048 sha256=28c1d76db0bffb1f0a11f81891d0fbb226f3fe96b638fa207ac10cfc120f9f43
  Stored in directory: /root/.cache/pip/wheels/d6/64/cd/32819b511a488e4993f2fab909a95330289c3f4e0f6ef4676d
Successfully built pingouin littleutils
  Attempting uninstall: statsmodels
    Found existing installation: statsmodels 0.12.2
    Uninstalling statsmodels-0.12.2:
      Successfully uninstalled statsmodels-0.12.2


In [ ]:
from pingouin import corr

In [ ]:
corr(df['Age_X'], df['Glucose_Level_Y'])

n         r          CI95%     p-val   BF10     power
pearson  6  0.529809  [-0.49, 0.94]  0.279645  0.814  0.199925

Interpretation: p-value (0.2796) is greater than 0.05 (level of significance). Therefore, the calculated correlation coefficient is not statistically significant at 5% level of significance.

Conclusion: There is no significant linear relationship between the two variables (Age and Glucose Level)

In [ ]:
!pip install researchpy

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
from researchpy.correlation import corr_case

In [ ]:
corr_case(df[['Age_X', 'Glucose_Level_Y']])

(  Pearson correlation test using list-wise deletion
 0                       Total observations used = 6,
                   Age_X Glucose_Level_Y
 Age_X               1.0          0.5298
 Glucose_Level_Y  0.5298             1.0,
                   Age_X Glucose_Level_Y
 Age_X            0.0000          0.2796
 Glucose_Level_Y  0.2796          0.0000)